### ISSUE TITLE
no case clause matching: :eacces

### ISSUE THREAD

**Issue by CraigCottingham** [nerves-project/nerves-examples#32]
Elixir 1.3.4
following the instructions in the README file

```
craigc@samantha:~/Projects/nerves-examples/blinky$ git pull
Already up-to-date.
craigc@samantha:~/Projects/nerves-examples/blinky$ mix deps.get
Running dependency resolution
Dependency resolution completed
  bbmustache: 1.0.4
  cf: 0.2.2
  erlware_commons: 0.22.0
  exrm: 1.0.8
  getopt: 0.8.2
  nerves: 0.3.4
  nerves_leds: 0.7.0
  nerves_system: 0.1.6
  nerves_system_br: 0.7.0
  nerves_system_rpi: 0.7.0
  nerves_toolchain: 0.7.1
  nerves_toolchain_armv6_rpi_linux_gnueabi: 0.7.2
  nerves_toolchain_ctng: 0.7.1
  providers: 1.6.0
  relx: 3.22.0
All dependencies up to date
craigc@samantha:~/Projects/nerves-examples/blinky$ mix firmware
Compiling 1 file (.ex)
Generated blinky app
Building release with MIX_ENV=dev.

You have dependencies (direct/transitive) which are not in :applications!
The following apps should be added to :applications in mix.exs:

        nerves_system_rpi -> nerves_system                                                     => nerves_system is missing from nerves_system_rpi
        nerves_system_rpi -> nerves_system_br                                                  => nerves_system_br is missing from nerves_system_rpi
        nerves_system_rpi -> nerves_toolchain_armv6_rpi_linux_gnueabi -> nerves_toolchain      => nerves_toolchain is missing from nerves_toolchain_armv6_rpi_linux_gnueabi
        nerves_system_rpi -> nerves_toolchain_armv6_rpi_linux_gnueabi -> nerves_toolchain_ctng => nerves_toolchain_ctng is missing from nerves_toolchain_armv6_rpi_linux_gnueabi
        nerves                                                                                 => nerves is missing from blinky

==> Modifying release for Nerves System rpi
** (CaseClauseError) no case clause matching: :eacces
    lib/exrm/utils/utils.ex:110: ReleaseManager.Utils.relx/5
    lib/mix/tasks/release.ex:339: anonymous fn/4 in Mix.Tasks.Release.do_release/1
    lib/ex_unit/capture_io.ex:146: ExUnit.CaptureIO.do_capture_io/2
    lib/ex_unit/capture_io.ex:119: ExUnit.CaptureIO.do_capture_io/3
    lib/mix/tasks/release.ex:338: Mix.Tasks.Release.do_release/1
    lib/mix/tasks/release.ex:78: Mix.Tasks.Release.do_run/1
    (mix) lib/mix/task.ex:296: Mix.Task.run_task/3
    lib/mix/tasks/firmware.ex:27: Mix.Tasks.Firmware.run/1
```

It looks like a permissions error, but on what?

**Comment by CraigCottingham** [comment]
Thanks to @jordan0day I have a workaround for this issue. Modify `mix.exs` for the project like so:
```
diff --git a/blinky/mix.exs b/blinky/mix.exs
index 346429a..179312f 100644
--- a/blinky/mix.exs
+++ b/blinky/mix.exs
@@ -24,7 +24,10 @@ defmodule Blinky.Mixfile do

   defp deps do
     [{:nerves, "~> 0.3.0"},
-     {:nerves_leds, "~> 0.7.0"}]
+     {:nerves_leds, "~> 0.7.0"},
+     {:relx, "3.11.0", override: true},
+     {:erlware_commons, "0.21.0", override: true}
+   ]
   end

   def system(target) do
```
Apparently there's a version mismatch in relx; I'm guessing it's that nerves doesn't like one that's too recent.

I'll be PRing this change soon.

**Comment by fhunleth** [comment]
@mobileoverlord - Hopefully you've seen this. I can't wait for your distillery work to get rid of the relx/erlware_commons issues, but what's the ETA on it? Should we merge @CraigCottingham's update in the mean time?

**Comment by mobileoverlord** [comment]
For backwards compatibility I think we should. The distillery branch is going to be merged soon, but will only work with the latest system_br and therefore the latest systems.

**Comment by CraigCottingham** [comment]
Should I add the same dependencies to the other projects as well?

**Comment by mobileoverlord** [comment]
we should patch `:nerves` so that it does this and put out a bug fix release. I am going to move this issue over there.

**Comment by GregMefford** [comment]
@mobileoverlord Is this still an issue now that Distillery support is merged in https://github.com/nerves-project/nerves/pull/59?

**Comment by mobileoverlord** [comment]
It isn't not. We should update the examples to pull in the latest versions

**Comment by GregMefford** [comment]
@CraigCottingham I believe this issue has been resolved by merging in #38.
Thanks again for taking the time to open this PR and please let us know if you still have any concerns! 🚀
